frontend/config-assistant/src/helpers/schema/schemaPreprocessor.ts

```diff
@@ -63,3 +79,15 @@ function hasRef(schema: JsonSchemaObjectType): boolean {
 function hasAllOfs(schema: JsonSchemaObjectType): boolean {
   return schema.allOf !== undefined && schema.allOf.length > 0;
 }
+function hasOneOfs(schema: JsonSchemaObjectType): boolean {
+  return schema.oneOf !== undefined && schema.oneOf.length > 0;
+}
+function hasAnyOfs(schema: JsonSchemaObjectType): boolean {
+  return schema.anyOf !== undefined && schema.anyOf.length > 0;
+}
+
+function optimizeSchema(schema: JsonSchemaObjectType) {
+  if (hasOneOfs(schema)) {
+    // TODO: if it is just oneOf of types, then replace it by a choice of types
```

Comment:
There is a lot we can possible do here, not just types. We could check if the oneOf schemas are even relevant for the GUI editor